Resources/ProtobufDumper/ProtobufDumper/ExecutableScanner.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace ProtobufDumper
+{
+    class ExecutableScanner
+    {
+        private static readonly Regex ProtoFileNameRegex = new Regex( @"^[a-zA-Z_0-9\\/.]+\.proto$", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.ExplicitCapture );
+
+        public delegate bool ProcessCandidate( string name, byte[] buffer );
+
+        public static void ScanFile( string fileName, ProcessCandidate processCandidate )
+        {
+            Console.WriteLine( "Loading binary '{0}'...", fileName );
+
+            ScanFile( File.ReadAllBytes( fileName ), processCandidate );
```

Comment:
I generally don't like using `ReadAllXXX` and loading everything into memory at once. Can we scan a `Stream` instead?

Reply:
This is not performant as a FileStream. I suspect ReadByte is the issue. This will require further investigation if we wanted to stream this and manage byte arrays.